### DISCUSSION TITLE
XCode 12.4 resulting in "Clang frontend command failed due to signal compilation error"

### DISCUSSION THREAD

**Discussion by JoshuaHalsted** [idaholab/moose#17850] [Q&A General] [answered]
Hello,

I am also having the same issues. However, when I uninstalled XCode 12.5 and installed XCode 12.4, I get essentially the same problem. I made sure that XCode was installed properly and was able to open the application.

I've attached the complete output message when I try to run the "make -j 4" command as specified in the tutorial.

Thanks,
Josh

Using HIT from /Users/joshhalsted/projects/moose/framework/contrib/hit
Compiling C++ (in opt mode) /Users/joshhalsted/projects/moose/framework/build/unity_src/linesearches_Unity.C...
In file included from /Users/joshhalsted/projects/moose/framework/build/unity_src/linesearches_Unity.C:2:
In file included from /Users/joshhalsted/projects/moose/framework/src/linesearches/LineSearch.C:10:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/LineSearch.h:12:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/MooseObject.h:13:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/InputParameters.h:13:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/MooseUtils.h:13:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/HashMap.h:13:
In file included from /Users/joshhalsted/opt/anaconda3/envs/moose/libmesh/include/libmesh/threads.h:24:
In file included from /Users/joshhalsted/opt/anaconda3/envs/moose/libmesh/include/libmesh/libmesh_common.h:36:
In file included from /Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:85:
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/stdlib.h:142:34: error: unknown type name 'ldiv_t'
inline _LIBCPP_INLINE_VISIBILITY ldiv_t div(long __x, long __y) _NOEXCEPT {
^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/stdlib.h:143:12: error: no member named 'ldiv' in the global namespace
return ::ldiv(__x, __y);
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/stdlib.h:146:34: error: unknown type name 'lldiv_t'
inline _LIBCPP_INLINE_VISIBILITY lldiv_t div(long long __x,
^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/stdlib.h:148:12: error: no member named 'lldiv' in the global namespace
return ::lldiv(__x, __y);
~~^
In file included from /Users/joshhalsted/projects/moose/framework/build/unity_src/linesearches_Unity.C:2:
In file included from /Users/joshhalsted/projects/moose/framework/src/linesearches/LineSearch.C:10:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/LineSearch.h:12:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/MooseObject.h:13:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/InputParameters.h:13:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/MooseUtils.h:13:
In file included from /Users/joshhalsted/projects/moose/framework/build/header_symlinks/HashMap.h:13:
In file included from /Users/joshhalsted/opt/anaconda3/envs/moose/libmesh/include/libmesh/threads.h:24:
In file included from /Users/joshhalsted/opt/anaconda3/envs/moose/libmesh/include/libmesh/libmesh_common.h:36:
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:99:9: error: no member named 'size_t' in the global namespace
using ::size_t;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib💯9: error: no member named 'div_t' in the global namespace
using ::div_t;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:101:9: error: no member named 'ldiv_t' in the global namespace
using ::ldiv_t;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:103:9: error: no member named 'lldiv_t' in the global namespace
using ::lldiv_t;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:105:9: error: no member named 'atof' in the global namespace
using ::atof;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:106:9: error: no member named 'atoi' in the global namespace
using ::atoi;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:107:9: error: no member named 'atol' in the global namespace
using ::atol;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:109:9: error: no member named 'atoll' in the global namespace
using ::atoll;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:111:9: error: no member named 'strtod' in the global namespace
using ::strtod;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:112:9: error: no member named 'strtof' in the global namespace
using ::strtof;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:113:9: error: no member named 'strtold' in the global namespace
using ::strtold;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:114:9: error: no member named 'strtol' in the global namespace
using ::strtol;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:116:9: error: no member named 'strtoll' in the global namespace
using ::strtoll;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:118:9: error: no member named 'strtoul' in the global namespace
using ::strtoul;
~~^
/Users/joshhalsted/opt/anaconda3/envs/moose/bin/../include/c++/v1/cstdlib:120:9: error: no member named 'strtoull' in the global namespace
using ::strtoull;
~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0. Program arguments: x86_64-apple-darwin13.4.0-clang++ -march=core2 -mtune=haswell -std=gnu++17 -DNDEBUG -Wall -Wextra -DADFPARSER_INCLUDES="-I/Users/joshhalsted/projects/moose/framework/include/utils -I/Users/joshhalsted/projects/moose/framework/include/base -I/Users/joshhalsted/opt/anaconda3/envs/moose/libmesh/include" -DMOOSE_TEST_ENABLED -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -std=c++14 -fmessage-length=0 -isystem /Users/joshhalsted/opt/anaconda3/envs/moose/include -O2 -felide-constructors -Qunused-arguments -Wunused-parameter -Wunused -fopenmp -DMETHOD=opt -Werror=return-type -Werror=reorder -Woverlength-strings -I/Users/joshhalsted/projects/moose/framework/build/header_symlinks -I/Users/joshhalsted/projects/moose/framework/contrib/boost/include -I/Users/joshhalsted/projects/moose/framework/contrib/json/include -I/Users/joshhalsted/projects/moose/framework/contrib/json/include/nlohmann -I/Users/joshhalsted/projects/moose/framework/contrib/mtwist/include -I/Users/joshhalsted/projects/moose/framework/contrib/pcre/include -I/Users/joshhalsted/projects/moose/framework/contrib/pugixml/include -I/Users/joshhalsted/projects/moose/framework/contrib/tinydir/include -I/Users/joshhalsted/projects/moose/framework/contrib/gtest -I/Users/joshhalsted/projects/moose/framework/contrib/hit -I/Users/joshhalsted/opt/anaconda3/envs/moose/libmesh/include -I/Users/joshhalsted/opt/anaconda3/envs/moose/libmesh-vtk/include/vtk-6.3 -D_THREAD_SAFE -pthread -I/Users/joshhalsted/opt/anaconda3/envs/moose/include -I/Users/joshhalsted/opt/anaconda3/envs/moose//include -I/Users/joshhalsted/projects/moose/test/build/header_symlinks -I/Users/joshhalsted/opt/anaconda3/envs/moose/libmesh/include -I/Users/joshhalsted/opt/anaconda3/envs/moose/libmesh-vtk/include/vtk-6.3 -D_THREAD_SAFE -pthread -I/Users/joshhalsted/opt/anaconda3/envs/moose/include -I/Users/joshhalsted/opt/anaconda3/envs/moose//include -MMD -MP -MF /Users/joshhalsted/projects/moose/framework/build/unity_src/linesearches_Unity.x86_64-apple-darwin13.4.0.opt.lo.d -MT /Users/joshhalsted/projects/moose/framework/build/unity_src/linesearches_Unity.x86_64-apple-darwin13.4.0.opt.lo -c /Users/joshhalsted/projects/moose/framework/build/unity_src/linesearches_Unity.C -fno-common -DPIC -o /Users/joshhalsted/projects/moose/framework/build/unity_src/.libs/linesearches_Unity.x86_64-apple-darwin13.4.0.opt.o -I/Users/joshhalsted/opt/anaconda3/envs/moose/include

parser at end of file
0 libLLVM-11.dylib 0x000000010d212c16 llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 54
1 libLLVM-11.dylib 0x000000010d211a08 llvm::sys::RunSignalHandlers() + 248
2 libLLVM-11.dylib 0x000000010d212142 llvm::sys::CleanupOnSignal(unsigned long) + 258
3 libLLVM-11.dylib 0x000000010d13a24e CrashRecoverySignalHandler(int) + 206
4 libsystem_platform.dylib 0x00007fff2053dd7d _sigtramp + 29
5 libclang-cpp.11.dylib 0x0000000109cde075 clang::EvaluatedExprVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter>::VisitStmt(clang::Stmt*) + 133
6 libclang-cpp.11.dylib 0x0000000109703c5f clang::ASTContext::getBaseElementType(clang::QualType) const + 47
7 libclang-cpp.11.dylib 0x0000000109cdd56a clang::StmtVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter, void>::Visit(clang::Stmt*) + 1418
8 libclang-cpp.11.dylib 0x0000000109cde075 clang::EvaluatedExprVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter>::VisitStmt(clang::Stmt*) + 133
9 libclang-cpp.11.dylib 0x0000000109cde075 clang::EvaluatedExprVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter>::VisitStmt(clang::Stmt*) + 133
10 libclang-cpp.11.dylib 0x0000000109cde075 clang::EvaluatedExprVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter>::VisitStmt(clang::Stmt*) + 133
11 libclang-cpp.11.dylib 0x0000000109cdcaef (anonymous namespace)::DeferredDiagnosticsEmitter::checkFunc(clang::SourceLocation, clang::FunctionDecl*) + 1695
12 libclang-cpp.11.dylib 0x0000000109cd2bbb clang::Sema::emitDeferredDiags() + 379
13 libclang-cpp.11.dylib 0x0000000109cd1e40 clang::Sema::ActOnEndOfTranslationUnitFragment(clang::Sema::TUFragmentKind) + 576
14 libclang-cpp.11.dylib 0x0000000109cd2e59 clang::Sema::ActOnEndOfTranslationUnit() + 345
15 libclang-cpp.11.dylib 0x00000001096e5de3 clang::Parser::ParseTopLevelDecl(clang::OpaquePtrclang::DeclGroupRef&, bool) + 1267
16 libclang-cpp.11.dylib 0x000000010962970d clang::ParseAST(clang::Sema&, bool, bool) + 525
17 libclang-cpp.11.dylib 0x000000010af0b5b8 clang::FrontendAction::Execute() + 104
18 libclang-cpp.11.dylib 0x000000010ae9c721 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) + 2145
19 libclang-cpp.11.dylib 0x000000010af7d06e clang::ExecuteCompilerInvocation(clang::CompilerInstance*) + 1998
20 x86_64-apple-darwin13.4.0-clang++ 0x00000001094d8d53 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) + 2131
21 x86_64-apple-darwin13.4.0-clang++ 0x00000001094d6e35 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) + 357
22 libclang-cpp.11.dylib 0x000000010ab376c7 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optionalllvm::StringRef >, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator >, bool) const::$_1>(long) + 23
23 libLLVM-11.dylib 0x000000010d139fda llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) + 218
24 libclang-cpp.11.dylib 0x000000010ab36a65 clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optionalllvm::StringRef >, std::__1::basic_string<char, std::__1::char_traits, std::__1::allocator >, bool) const + 405
25 libclang-cpp.11.dylib 0x000000010ab03cf1 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&) const + 433
26 libclang-cpp.11.dylib 0x000000010ab0426c clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::__1::pair<int, clang::driver::Command const*> >&) const + 140
27 libclang-cpp.11.dylib 0x000000010ab1ddec clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::__1::pair<int, clang::driver::Command const*> >&) + 428
28 x86_64-apple-darwin13.4.0-clang++ 0x00000001094d67c8 main + 10264
29 libdyld.dylib 0x00007fff20513f3d start + 1
clang-11: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 11.0.0
Target: x86_64-apple-darwin13.4.0
Thread model: posix
InstalledDir: /Users/joshhalsted/opt/anaconda3/envs/moose/bin
clang-11: note: diagnostic msg:
PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-11: note: diagnostic msg: /var/folders/qr/b09rm3rs6hz9c18pp3_301jm0000gn/T/linesearches_Unity-9916ec.cpp
clang-11: note: diagnostic msg: /var/folders/qr/b09rm3rs6hz9c18pp3_301jm0000gn/T/linesearches_Unity-9916ec.sh
clang-11: note: diagnostic msg: Crash backtrace is located in
clang-11: note: diagnostic msg: /Users/joshhalsted/Library/Logs/DiagnosticReports/clang-11__.crash
clang-11: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
clang-11: note: diagnostic msg:

make: *** [/Users/joshhalsted/projects/moose/framework/build/unity_src/linesearches_Unity.x86_64-apple-darwin13.4.0.opt.lo] Error 1
(moose) Joshs-MacBook-Air:test joshhalsted$

**Comment by JoshuaHalsted** [comment, marked as answer]
I was able to resolve the problem. I had to dig around my drive using OmniDiskSweeper to find any evidence that a previous version of XCode was being used (even though I had uninstalled it and the CommandLineTools prior). There was a folder called "Xcode" in the /Library/Developer/CommandLineTools path, which I deleted (I didn't previously) and, upon full installation of 12.4, was able to successfully run the tests.

**Reply by milljm** [reply]
This is good to know! Thank you for posting!
